app/models/store.rb

```diff
@@ -1,7 +1,24 @@
 class Store < ApplicationRecord
-
+  has_many :user_roles
+  has_many :users, through: :user_roles
   has_many :items
 
+  before_validation :generate_url
+
   enum status: ["pending", "active", "suspended"]
 
+  def self.create_with_user(store_info, user)
+    create(store_info).associate_user(user)
+  end
+
+  def associate_user(user)
```

Comment:
I just spoke with @adrian-lara about this method because this initially looked like it was outside the scope of this card (I was asking "wouldn't a user already be a registered user, so why would we have to add another record in the user_roles column").

I'm adding this comment for documentation purposes.

This method creates another record in the user_roles table so that this new store can be associated with the user since a user has many stores through the user_roles table.